src/rt/test/func/bag/bag.cc

```diff
@@ -0,0 +1,95 @@
+// Copyright Microsoft and Project Verona Contributors.
+// SPDX-License-Identifier: MIT
+#include "ds/bag.h"
+
+#include "set"
+#include "test/harness.h"
+#include "test/log.h"
+#include "verona.h"
+
+using namespace snmalloc;
+using namespace verona::rt;
+
+void test()
+{
+  auto& alloc = ThreadAlloc::get();
+
+  Bag<uintptr_t, uintptr_t, Alloc> bag;
+
+  const int NUM_OBJECTS = 127;
+
+  // Allocate a bunch of objects so that the region vector uses many
+  // allocation blocks to track each object in the region.
+  for (uintptr_t i = 0; i < NUM_OBJECTS; i++)
+  {
+    bag.insert({nullptr, i}, alloc);
+  }
+
+  // Iterate through the bag and make sure that all the
+  // entries make sense.
+  Bag<uintptr_t, uintptr_t, Alloc>::iterator iter(&bag);
+
+  Bag<uintptr_t, uintptr_t, Alloc>::Elem* rm1 = nullptr;
+  Bag<uintptr_t, uintptr_t, Alloc>::Elem* rm2 = nullptr;
+  Bag<uintptr_t, uintptr_t, Alloc>::Elem* rm3 = nullptr;
+
+  uintptr_t count = 0;
+  for (auto entry : iter)
+  {
+    check(entry->metadata == (NUM_OBJECTS - count - 1));
+
+    if (count == 5)
+      rm1 = entry;
+
+    if (count == 40)
+      rm2 = entry;
+
+    if (count == 100)
+      rm3 = entry;
+    count++;
+  }
+
+  check(count == NUM_OBJECTS);
+
+  std::set<uintptr_t> removed = {
```

Comment:
If you add the shorthands above, then this can be:
```suggestion
  std::set<E> removed = {
```
and then you can drop all the casts to `uintptr_t`.  Or is this because you need a less defined on the type?  Does `unordered_set` help here?  You aren't traversing in an order anyway?

Reply:
I need a less defined type here as I just want to compare the pointer addresses. I've changed it to an `unordered_set` though: 85ae978cb22f4acb0bf1ffd956f62d9e95b4282f